File: lib/models/media.dart
class Media {
  final String url;
  final MediaFormat type;

  const Media({required this.type, required this.url});

  Map<String, String> toMap() {
    return <String, String>{
      "url": url,
      "fileType": type.fileExtension,
    };
  }
}

abstract class MediaFormat {
  final String fileExtension;
  const MediaFormat(this.fileExtension);

  bool get isAudioFormat;
  bool get isVideoFormat;
}

class AudioFormat extends MediaFormat {
  AudioFormat.mp3() : super("mp3");
  AudioFormat.m4a() : super("m4a");
  AudioFormat.wav() : super("wav");

  @override
  bool get isAudioFormat => true;

  @override
  bool get isVideoFormat => false;
}

class VideoFormat extends MediaFormat {
  VideoFormat.mp4() : super("mp4");
  VideoFormat.mkv() : super("mkv");
  VideoFormat.webm() : super("webm");

  @override
  bool get isAudioFormat => false;

  @override
  bool get isVideoFormat => true;
}

File: android/src/main/kotlin/com/okellogerald/av_player/Media.kt
package com.okellogerald.av_player

import androidx.media3.common.MimeTypes
import java.io.Serializable

data class Media constructor(
    val url: String,
    val fileType: String,
) : Serializable {
    companion object {
        fun network(url: String, fileType: String) = Media(url, fileType)
    }

    fun getMimeType(): String? {
        return when (fileType) {
            "mp4" -> {
                MimeTypes.VIDEO_MP4
            }

            "mkv" -> {
                MimeTypes.VIDEO_H263
            }

            "webm" -> {
                MimeTypes.VIDEO_WEBM
            }

            "wav" -> {
                MimeTypes.AUDIO_WAV
            }

            "mp3" -> {
                MimeTypes.AUDIO_AC4
            }

            "m4a" -> {
                MimeTypes.AUDIO_AAC
            }

            else -> null
        }
    }
}